flocker/control/test/test_protocol.py

```diff
@@ -655,3 +655,32 @@ def test_logging(self, logger):
                 "agent": connection_protocol,
             }
         )
+
+    @validate_logging(None)
+    def test_error_sending(self, logger):
+        """
+        An error sending to one agent does not prevent others from being
+        notified.
+        """
+        control_amp_service = build_control_amp_service(self)
+        self.patch(control_amp_service, 'logger', logger)
+
+        connected_protocol = ControlAMP(control_amp_service)
+        connected_protocol.callRemote = lambda *args, **kwargs: succeed({})
+
+        error = ConnectionLost()
+        disconnected_protocol = ControlAMP(control_amp_service)
+        results = [succeed({}), fail(error)]
+        disconnected_protocol.callRemote = (
+            lambda *args, **kwargs: results.pop(0))
```

Comment:
There's `flocker.testtools.FakeAMPClient` which is probably marginally better than this.  It would also seem reasonable to have an `AlwaysSucceedAMPClient` and an `AlwaysFailAMPClient` that implement `callRemote` in the ways implied by their names.  (As far as I can tell nothing about `control_amp_service.connected` cares what kind of `AMP` subclass is passed to it.)  Can be a follow-up ticket if you want. If so link to it from here.